### ISSUE TITLE
Exams: Validate exam authorizations were accepted by Pearson

### ISSUE THREAD

**Issue by rhysyngsun** [mitodl/micromasters#2080]
This validates that exam authorizations sent by the mechanism for #2076 were successfully processed and acknowledged by Pearson. The files are referred to as Exam Authorization Confirmations (EAC). Upon validation of acceptance it would then be possible for the user to SSO over to the Pearson site and complete exam scheduling for the given exam.

- [x] Download the EAC files from the SFTP server
- [x] Process the EAC files for success/failures
  - With a success, user should be ready to SSO and complete exam scheduling
  - With a failure, email sent to `MICROMASTERS_ADMIN_EMAIL`
- [x] Remove processed files from SFTP server

Open Questions:

- [x] What do we want to trigger on failure of an exam authorization? Probably an email to someone?

**Comment by rhysyngsun** [comment]
@pdpinch questions over here (not immediate priority)

**Comment by pdpinch** [comment]
For exam authorization errors, I would suggest 1) raising an exception so that it's caught by Sentry and logged and 2) e-mail the address in MICROMASTERS_ADMIN_EMAIL

**Comment by amir-qayyum-khan** [comment]
@pdpinch @rhysyngsun 

- How many records will be in one EAC file?
- Currently i am sending summery of failures for each file as email to ``MICROMASTERS_ADMIN_EMAIL``, is this approach right?

**Comment by pdpinch** [comment]
1. Why does the number of records matter? The number of records in an EAC file should correspond to the number in the EAD file. For the pilot, this could be as many as 4500, although probably will be much less. For the May exam offering, it looks like it could be around 3000. 
2. Sending the summary of failures to MICROMASTERS_ADMIN_EMAIL sounds good. Are they also logged somehow?
3. I defer to @rhysyngsun but I think we'd want to keep the file around in the case of any error. We aren't worried about filename collisions, right? 
4. @rhysyngsun can confirm, but I think we will only get a response from Pearson once every 24 hours.

**Comment by amir-qayyum-khan** [comment]
thanks @pdpinch 

> Why does the number of records matter? The number of records in an EAC file should correspond to the number in the EAD file. For the pilot, this could be as many as 4500, although probably will be much less. For the May exam offering, it looks like it could be around 3000.

This was for my information.

> Sending the summary of failures to MICROMASTERS_ADMIN_EMAIL sounds good. Are they also logged somehow?

Yes i am logging failure.

**Comment by rhysyngsun** [comment]
> 3. I defer to @rhysyngsun but I think we'd want to keep the file around in the case of any error. We aren't worried about filename collisions, right?

@amir-qayyum-khan @pdpinch depends on what we mean by keeping the file around...do we want to persist a copy to S3? We can't store them locally on the dynos and I would expect Pearson periodically evicts these files . Additionally, I discovered yesterday right before I had to leave that the zip file that has the EAC file(s) _also_ contains the VCDC files (this nuance wasn't documented), which adds its own complication and necessitates a redesign of the core code here (currently working on this). Or we could leave the file but continue processing and we'll have to reprocess everything again later once the issue is fixed.

> So their should be a task which will process these EAC files, After what interval it should look for files on ftp?

I think it's up to twice a day. I already have a ticket to add these to the crontab in #2426
